### PR TITLE
Add fmt::format_as for SpillStats and MemoryArbitrator::Stats

### PR DESCRIPTION
**Pull request by majetideepak**

### CODE REVIEW

**Review by mbasmanova — COMMENTED** [review]
Let's a test that fails without this change.

File: velox/common/base/SpillStats.h [hunk 1]
```diff
@@ -141,3 +141,13 @@ void updateGlobalMaxSpillLevelExceededCount(
 /// Gets the cumulative global spill stats.
 SpillStats globalSpillStats();
 } // namespace facebook::velox::common
+
+template <>
```
Comment (kgpai) [inline velox/common/base/SpillStats.h:145]: As we have moved to fmt 10 anyway, why not just use format_as and do away with the template specialization ?

**Comment by majetideepak** [comment]
> Let's a test that fails without this change.

@mbasmanova The error only happens when we use `VELOX_CHECK_*` checks as they use fmt internally.
Do you propose I add a check in the tests even though we use gtest checks?

**Comment by majetideepak** [comment]
I could invoke fmt I suppose as `ASSERT_EQ(zeroStats, stats1) << fmt::format("Failed {}!={}", zeroStats, stats1);`

**Comment by mbasmanova** [comment]
I assume a test that applies fmt::format to a Stats object would be sufficient, no?

File: velox/common/base/SpillStats.h (hunk 1 above)
Comment (majetideepak) [reply velox/common/base/SpillStats.h:145]: good point!

File: velox/common/base/SpillStats.h [hunk 2]
```diff
@@ -140,4 +140,9 @@ void updateGlobalMaxSpillLevelExceededCount(
 
 /// Gets the cumulative global spill stats.
 SpillStats globalSpillStats();
+
+/// Formatter for fmt.
+auto format_as(SpillStats f) {
```
Comment (mbasmanova) [inline velox/common/base/SpillStats.h:145]: f -> stats

auto -> std::string

Using 'auto' for return types makes it very hard to read the code.

File: velox/common/memory/MemoryArbitrator.h [hunk 3]
```diff
@@ -243,6 +243,11 @@ class MemoryArbitrator {
   const bool checkUsageLeak_;
 };
 
+/// Formatter for fmt.
+auto format_as(MemoryArbitrator::Stats f) {
```
Comment (mbasmanova) [inline velox/common/memory/MemoryArbitrator.h:247]: ditto

**Review by mbasmanova — COMMENTED** [review]
Looks good % some nits. Thank you for adding tests. Would you confirm that these tests fail without code changes?

**Comment by majetideepak** [comment]
> Would you confirm that these tests fail without code changes?

yes, I confirmed that the tests fail to build without the `format_as`. @kgpai is confirming as well.

**Comment by kgpai** [comment]
@majetideepak I am getting the following error in our internal builds: 
``` 
 static assertion failed due to requirement 'formattable': Cannot format an argument. To make type T formattable provide a formatter<T> specialization: https://fmt.dev/latest/api.html#udt
  static_assert(
  ^
buck-out/v2/gen/fbsource/29760d6803037a85/third-party/fmt/___fmtFbcode__/buck-headers/fmt/core.h:1786:10: note: in instantiation of function template specialization 'fmt::detail::make_value<fmt::basic_format_context<fmt::appender, char>, facebook::velox::common::SpillStats &>' requested here
  return make_value<Context>(val);
         ^
buck-out/v2/gen/fbsource/29760d6803037a85/third-party/fmt/___fmtFbcode__/buck-headers/fmt/core.h:1908:23: note: in instantiation of function template specialization 'fmt::detail::make_arg<true, fmt::basic_format_context<fmt::appender, char>, fmt::detail::type::custom_type, facebook::velox::common::SpillStats &, 0>' requested here
        data_{detail::make_arg<
                      ^
buck-out/v2/gen/fbsource/29760d6803037a85/third-party/fmt/___fmtFbcode__/buck-headers/fmt/core.h:1927:10: note: in instantiation of function template specialization 'fmt::format_arg_store<fmt::basic_format_context<fmt::appender, char>, facebook::velox::common::SpillStats>::format_arg_store<facebook::velox::common::SpillStats &>' requested here
  return {FMT_FORWARD(args)...};
         ^
buck-out/v2/gen/fbsource/29760d6803037a85/third-party/fmt/___fmtFbcode__/buck-headers/fmt/core.h:3215:28: note: in instantiation of function template specialization 'fmt::make_format_args<fmt::basic_format_context<fmt::appender, char>, facebook::velox::common::SpillStats &>' requested here
  return vformat(fmt, fmt::make_format_args(args...));
                           ^
velox/common/base/tests/SpillStatsTest.cpp:109:12: note: in instantiation of function template specialization 'fmt::format<facebook::velox::common::SpillStats &>' requested here
      fmt::format("{}", stats2),
     ``` 
     I am afraid i might have to use the template specialization..

**Review by xiaoxmeng — APPROVED** [review]
@kgpai thanks!

**Comment by majetideepak** [comment]
> I am getting the following error in our internal builds:

@kgpai I believe Meta build is using fmt-9 which probably does not support format_as. Let's stick to the fmt::formatter.

**Comment by majetideepak** [comment]
But the documentation https://fmt.dev/9.0.0/api.html does have format_as. Strange. We can look into this in a separate PR.

**Comment by kgpai** [comment]
The curious thing is it is only happening for SpillStats and not MemoryArbitor. I will investigate and create a subsequent PR but do not want to block this PR and the presto build.

**Pull request closed without merging**